core/defaults/src/main/java/org/eclipse/dataspaceconnector/core/defaults/negotiationstore/InMemoryContractNegotiationStore.java

```diff
@@ -25,92 +26,41 @@
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.Comparator;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
 import java.util.Objects;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
-import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-import static java.util.stream.Collectors.toList;
-
 /**
  * An in-memory, threadsafe process store. This implementation is intended for testing purposes only.
  */
-public class InMemoryContractNegotiationStore implements ContractNegotiationStore {
+public class InMemoryContractNegotiationStore extends InMemoryStateMachineStore<ContractNegotiation> implements ContractNegotiationStore {
```

Comment:
What's the reason to use inheritance for `InMemoryStateMachineStore` instead of composition? It adds more complexity to the code and shouldn't be use to make the code reusable.
InMemoryStateMachineStore could be just a field in the specific stores:
```java
    private InMemoryStateMachineStore<ContractNegotiation> store = new InMemoryStateMachineStore<>(ContractNegotiation.class);
```

Reply:
Excellent observation @Izzzu. Fixed accordingly